digits/tools/create_db.py

```diff
@@ -693,7 +725,10 @@ def _save_means(image_sum, image_count, mean_files):
             with open(mean_file, 'wb') as outfile:
                 outfile.write(blob.SerializeToString())
         elif mean_file.lower().endswith(('.jpg', '.jpeg', '.png')):
-            image = PIL.Image.fromarray(mean)
+            if np.issubdtype(mean.dtype, np.float):
+                image = PIL.Image.fromarray(mean*255/mean.max()).convert('L')
```

Comment:
You are right.  I don't think any pixel in mean can be greater than 65535.  We should be able to do that for PNG case.